### PR TITLE
Configurable timeout for CDP requests

### PR DESCRIPTION
**Pull request by art-alexeyenko**
Adds an env variable to set CDP request timeout

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/create-sitecore-jss/src/templates/nextjs-personalize/.env [hunk 1]
```diff
@@ -12,3 +12,6 @@ NEXT_PUBLIC_CDP_SCRIPT_URL=https://d1mj578wat5n4o.cloudfront.net/boxever-1.4.8.m
 
 # Your Sitecore CDP point of sale
 NEXT_PUBLIC_CDP_POINTOFSALE=
+
+# Timeout for CDP requests to respond within (ms)
+PERSONALIZE_MIDDLEWARE_TIMEOUT=500
```
Comment (sc-addypathania) [inline packages/create-sitecore-jss/src/templates/nextjs-personalize/.env:17]: Did we decide the default value to be 500?
I remember in some situations it took us more than 1000ms.

Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs-personalize/.env:17]: Yes, we might want to bump this to 10000. For local development we were seeing higher response times. 
Also, worth considering, should we introduce a `.env.development` file ([Next.js supports these](https://nextjs.org/docs/basic-features/environment-variables#default-environment-variables)) and add this here instead (so it only applies in dev)?

**Review by ambrauer — CHANGES_REQUESTED** [review]

File: packages/create-sitecore-jss/src/templates/nextjs-personalize/.env (hunk 1 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/templates/nextjs-personalize/.env]: set dev value for 10k now

File: packages/create-sitecore-jss/src/templates/nextjs-personalize/src/lib/middleware/plugins/personalize.ts [hunk 2]
```diff
@@ -20,6 +20,8 @@ class PersonalizePlugin implements MiddlewarePlugin {
         endpoint: process.env.NEXT_PUBLIC_CDP_API_URL || '',
         clientKey: process.env.NEXT_PUBLIC_CDP_CLIENT_KEY || '',
         pointOfSale: process.env.NEXT_PUBLIC_CDP_POINTOFSALE || '',
+        timeout:
```
[the remark this answers is not in the document]
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/templates/nextjs-personalize/src/lib/middleware/plugins/personalize.ts:27]: edge endpoint differs from CDP endpoint - so I added an extra env value

File: packages/create-sitecore-jss/src/templates/nextjs-personalize/.env (hunk 1 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/templates/nextjs-personalize/.env]: done

File: packages/create-sitecore-jss/src/templates/nextjs-personalize/.env.production [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Timeout for CDP requests to respond within (ms)
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs-personalize/.env.production:1]: Do we need the .env.production file? We already have a fallback of 250 in the personalze.ts file and in the SDK middleware itself. We can leave addl production config up to their deployment platform of choice (e.g. set in Vercel project).

**Review by ambrauer — APPROVED** [review]
See some final suggestions, otherwise looks good.

File: packages/create-sitecore-jss/src/templates/nextjs-personalize/.env.production (hunk 3 above)
Comment (art-alexeyenko) [reply packages/create-sitecore-jss/src/templates/nextjs-personalize/.env.production:1]: Yes - if we want it to be configurable. If variables are put into regular env file that would be the value used over others. So I've put both dev and prod env files to differentiate between usage scenarios.

Comment (ambrauer) [reply packages/create-sitecore-jss/src/templates/nextjs-personalize/.env.production:1]: Hmm... I guess I'm just leery of introducing more of these .env files which start dictating a customer's setup for prod. 

Maybe it's best we just go back to how you had it originally in our single .env file and add a comment that these should be tuned down for production? This falls more in line with other variables which need to be adjusted per environment (e.g. PUBLIC_URL).

**Pull request merged**